parser_with_temp.py

```diff
@@ -0,0 +1,258 @@
+
+When dictionary is completed module creates an .xlsx file where inserts data
+from the dictionary in certain order.
+"""
+
+from bs4 import BeautifulSoup
+from datetime import datetime
+from os import remove
+from time import time
+from urlparse import urlparse, parse_qs
+import ast
+import re
+import requests
+import sys
+import xlsxwriter
+
+
+class Session(requests.Session):
+    """Class inherited from requests.Session object and has additional methods
+    """
+
+    def login(self, site_url, authentication_data):
+        """Sends a request for getting response with a token from the server
+        creates session properties containing base url and url with token for
+        future usage"""
+        try:
+            self.base_url = site_url
+            self.logined_url = self.post(site_url, authentication_data).url
+        except requests.adapters.ConnectionError:
+            sys.exit(
+                'Connection issues. Check connection and try again')
+
+
+    def get_token(self):
+        """parses token from the response and creates session token property"""
+        try:
+            self.token = '&token=' + parse_qs(urlparse(
+                self.logined_url).query).get('token')[0]
+        except TypeError:
+            sys.exit(
+                'Login issues. Please enter properly username and password.')
+
+    def get_top_number_of_general_page(self):
+        """ sends request for the first page containing general table and grabs
+        the top number of pages containing general tables. Applies
+        top_page_number to the session"""
+        route = 'index.php?route=sale/order'
+        general_table_page_response = self.post(
+            self.base_url + route + self.token)
+        general_table_page = BeautifulSoup(
+            general_table_page_response.text.encode('utf-8'), "html.parser")
+        pages_urls = general_table_page.find('div', {'class': 'pagination'})
+        page_numbers = []
+        for url_item in pages_urls.findAll('a'):
+            page_numbers.append(int(re.split(r'=', (
+                url_item.get('href')))[-1]))
+        self.top_page_number = max(page_numbers)
+
+    def get_id_list(self):
+        """Iterates a list up to top_page_number, calling pages with general
+        tables and grabs all order id's into a list with all orders id's
+        :return a list with all orders numbers.
+        """
+        route = 'index.php?route=sale/order'
+        page = '&page='
+        _id_list = []
+        for page_number in range(self.top_page_number):
+            table_page = BeautifulSoup(self.post(
+                self.base_url + route + self.token + page + str(
+                    page_number + 1)).text.encode('utf-8'), "html.parser")
+            table_body = table_page.find('table', {'class': 'list'}).find(
+                'tbody')
+            for row in table_body.findAll('tr'):
+                if row.find('input').get('value') != '':
```

Comment:
Yes, it's exactly what is needed. If this field doesn't exists on the page, it must be empty field in the resulting table.